### PR TITLE
Initial PR To review broker sdk changes thus far.

### PR DESCRIPTION
**Pull request by drew887**
Currently has most of the Accounts routes implemented and tested. CIP routes I was having issues with accessing those routes in the sandbox environment to verify so they are currently left unfinished with the goal of returning to them later.

### CODE REVIEW

File: alpaca/broker/client.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        super().__init__(api_key, secret_key, api_version, base_url, sandbox, raw_data)
+
+    # ############################## ACCOUNTS/TRADING ACCOUNTS ################################# #
+
+    def create_account(self, account_data: AccountCreationRequest) -> Account:
+        """
+        Create an account.
+
+        Args:
+            account_data(AccountCreationRequest): The data representing the Account you wish to create
```
Comment (haxdds) [inline alpaca/broker/client.py:74]: There should be a space between variable name and variable type
`account_data (AccountCreationRequest): The data representing the Account you wish to create`

File: alpaca/broker/client.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        return Account(**response)
+
+    def get_account_by_id(self, account_id: Union[UUID, str]) -> Account:
+        """
+        Get an Account by its associated account_id.
+
+        Note: If no account is found the api returns a 401, not a 404
+
+        Args:
+            account_id(Union[UUID,str]): The id of the account you wish to get. str uuid values will be upcast
```
Comment (haxdds) [inline alpaca/broker/client.py:93]: A space is needed here again.

File: alpaca/common/rest.py [hunk 3]
```diff
@@ -150,6 +151,8 @@ def _one_request(self, method: str, url: str, opts: dict, retry: int) -> dict:
                 error = resp.json()
                 if "code" in error:
                     raise APIError(error, http_error)
+            else:
+                raise
```
Comment (haxdds) [inline alpaca/common/rest.py:155]: Why the need for an empty raise?

File: alpaca/broker/enums.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    """
+    Represents what side this order was executed on.
+    """
+
+    BUY = "buy"
+    SELL = "sell"
+
+
+class OrderStatus(str, Enum):
+    NEW = "new"
```
Comment (haxdds) [inline alpaca/broker/enums.py:302]: Should we add a docstring here?

File: alpaca/broker/client.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        """
+
+        account_id = validate_account_id_param(account_id)
+
+        self.delete(f"/accounts/{account_id}")
+
+    def list_accounts(
+        self, search_parameters: Optional[ListAccountsRequest] = None
+    ) -> List[Account]:
+        # this is a get request, so we're safe not checking to see if we specified at least one param
```
Comment (haxdds) [inline alpaca/broker/client.py:155]: This method is missing docstrings.

File: alpaca/broker/client.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+        response = self.get(
+            f"/accounts",
+            params,
+        )
+
+        return parse_obj_as(List[Account], response)
+
+    def get_trade_account_by_id(self, account_id: Union[UUID, str]) -> TradeAccount:
+        account_id = validate_account_id_param(account_id)
```
Comment (haxdds) [inline alpaca/broker/client.py:170]: also needs docstrings.

File: alpaca/broker/models/accounts.py [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+        previous_close (Optional[datetime]): Previous sessions close time
+        last_long_market_value (Optional[str]): Value of all long positions as of previous trading day at 16:00:00 ET
+        last_short_market_value (Optional[str]): Value of all short positions as of previous trading day at 16:00:00 ET
+        last_cash (Optional[str]): Value of all cash as of previous trading day at 16:00:00 ET
+        last_initial_margin (Optional[str]): Value of initial_margin as of previous trading day at 16:00:00 ET
+        last_regt_buying_power (Optional[str]): Value of regt_buying_power as of previous trading day at 16:00:00 ET
+        last_daytrading_buying_power (Optional[str]): Value of daytrading_buying_power as of previous trading day at 16:00:00 ET
+        last_daytrade_count (Optional[int]): Value of daytrade_count as of previous trading day at 16:00:00 ET
+        last_buying_power (Optional[str]): Value of buying_power as of previous trading day at 16:00:00 ET
+        clearing_broker (Optional[ClearingBroker]):
```
Comment (haxdds) [inline alpaca/broker/models/accounts.py:371]: clearing_broker is missing a description

File: alpaca/broker/models/cip.py [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+        created_at (Optional[datetime]): Datetime for when this check was done
+        date_of_birth (Optional[datetime]): DOB for applicant if found on document
+        date_of_expiry (Optional[datetime]): date of expiry for the checked document
+        document_numbers (Optional[List[str]]): Number of the document that was checked
+        document_type (Optional[str]): Type of the document that was checked
+        first_name (Optional[str]): First name extracted from the document
+        last_name (Optional[str]): Last name extracted from the document
+        gender (Optional[str]): Gender info extracted from the document
+        issuing_country (Optional[str]): Country for which issued the document
+        nationality (Optional[str]):
```
Comment (haxdds) [inline alpaca/broker/models/cip.py:74]: nationality is missing a description

File: alpaca/broker/models/cip.py [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+
+
+class CIPIdentity(BaseModel):
+    """
+    Represents the results of running an identity check for a CIPInfo
+
+    Attributes:
+        id (str): Your internal ID of check
+        result (Optional[CIPResult]): Overall result of check
+        status (Optional[CIPStatus]): OVerall status of check
```
Comment (haxdds) [inline alpaca/broker/models/cip.py:163]: Small typo with OVerall

File: docs/api_reference/broker_client.rst [hunk 10]
```diff
@@ -0,0 +1,10 @@
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents:
+
+Broker API Client
+===================
```
Comment (haxdds) [inline docs/api_reference/broker_client.rst:6]: Really small thing, but we should have the underline as long as the text.

File: docs/api_reference/broker_enums.rst [hunk 11]
```diff
@@ -0,0 +1,11 @@
+.. toctree::
+   :maxdepth: 2
+   :caption: Contents:
+
+Broker API Enums
+===================
```
Comment (haxdds) [inline docs/api_reference/broker_enums.rst:6]: Again tiny change but we should the underline as the same length as the title.

**Review by haxdds — CHANGES_REQUESTED** [review]
A few things

File: alpaca/common/rest.py (hunk 3 above)
Comment (drew887) [reply alpaca/common/rest.py:155]: oh you're right I think I wanted this to be `raise http_error` so we don't swallow any `HTTPError`s :dizzy_face:

File: alpaca/broker/enums.py (hunk 4 above)
Comment (drew887) [reply alpaca/broker/enums.py:302]: Yes, this model is only temporary here anywho as it should probably be moved to the common module. I don't think broker has any unique order status' ? :thinking:

**Review by haxdds — APPROVED** [review]

**Pull request merged**